internal/cmd/backup.go

```diff
@@ -377,122 +393,39 @@ func backupRestoreCmdFunc(cmd *cobra.Command, args []string) error {
-				return fmt.Errorf("error sending batch to server: %w", errors.Join(err, closeErr))
-			}
+	batchesPerTransaction := cobrautil.MustGetUint(cmd, "batches-per-transaction")
 
-			// Reset the relationships in the batch
-			batch = batch[:0]
-			batchesWritten++
+	strategy := MustGetEnum[ConflictStrategy](cmd, "conflict-strategy", conflictStrategyMapping)
+	disableRetries := cobrautil.MustGetBool(cmd, "disable-retries")
+	requestTimeout := cobrautil.MustGetDuration(cmd, "request-timeout")
 
-			if batchesWritten%batchesPerTransaction == 0 {
-				resp, err := relationshipWriter.CloseAndRecv()
-				if err != nil {
-					return fmt.Errorf("error finalizing write of %d batches: %w", batchesPerTransaction, err)
-				}
-				written += int64(resp.NumLoaded)
-				if err := bar.Set64(written); err != nil {
-					return fmt.Errorf("error incrementing progress bar: %w", err)
-				}
-
-				if !isatty.IsTerminal(os.Stderr.Fd()) {
-					log.Trace().
-						Int64("batches", batchesWritten).
-						Int64("relationships", written).
-						Msg("restore progress")
-				}
-
-				relationshipWriter, err = client.BulkImportRelationships(ctx)
-				if err != nil {
-					return fmt.Errorf("error creating new writer stream: %w", err)
-				}
-			}
-		}
-	}
-
-	// Write the last batch
-	if err := relationshipWriter.Send(&v1.BulkImportRelationshipsRequest{
-		Relationships: batch,
-	}); err != nil {
-		return fmt.Errorf("error sending last batch to server: %w", err)
-	}
+	return newRestorer(schema, decoder, c, prefixFilter, batchSize, batchesPerTransaction, strategy,
+		disableRetries, requestTimeout).restoreFromDecoder(cmd.Context())
+}
 
-	// Finish the stream
-	resp, err := relationshipWriter.CloseAndRecv()
+// MustGetEnum is a helper for getting an enum value from a string cobra flag.
+func MustGetEnum[E constraints.Integer](cmd *cobra.Command, name string, mapping map[string]E) E {
+	value, err := cmd.Flags().GetString(name)
 	if err != nil {
-		return fmt.Errorf("error finalizing last write: %w", err)
-	}
-	batchesWritten++
-	written += int64(resp.NumLoaded)
-	if err := bar.Set64(written); err != nil {
-		return fmt.Errorf("error incrementing progress bar: %w", err)
+		panic("failed to find cobra flag: " + name)
 	}
-	totalTime := time.Since(relationshipWriteStart)
 
-	if err := bar.Finish(); err != nil {
-		return fmt.Errorf("error finalizing progress bar: %w", err)
+	value = strings.TrimSpace(strings.ToLower(value))
+	if enum, ok := mapping[value]; ok {
+		return enum
 	}
 
-	log.Info().
-		Int64("batches", batchesWritten).
-		Int64("relationships", written).
-		Uint64("perSecond", perSec(uint64(written), totalTime)).
-		Stringer("duration", totalTime).
-		Msg("finished restore")
-
-	return nil
-}
-
-func perSec(i uint64, d time.Duration) uint64 {
-	secs := uint64(d.Seconds())
-	if secs == 0 {
-		return i
-	}
-	return i / secs
+	panic(fmt.Sprintf("unexpected flag value %s: should be one of %v", value, maps.Keys(mapping)))
```

Comment:
I get what you mean - we use `Must*` so that it lets us know when we've missed wiring up something in the code about a required flag. Typically it means we have a default value.

I'm ok turning this into an error, given it's better UX, but it's in conflict with the fact it needs to call `MustGetString` and thus needs to be named `Must*` too.

Reply:
addressed in https://github.com/authzed/zed/pull/316/commits/2028f7c23fcef35f2794fd14faa009e2a47c901d